pkg/controller/cassandradatacenter/controller.go

```diff
@@ -161,83 +260,77 @@ func (r *CassandraDataCenterReconciler) createOrUpdateNodesService(cdc *cassandr
 			Selector: dataCenterLabels(cdc),
 		}
 
+		if err := controllerutil.SetControllerReference(cdc, nodesService, r.scheme); err != nil {
+			return err
+		}
+
 		return nil
 	})
 	if err != nil {
 		return nil, err
 	}
 
-	if err := controllerutil.SetControllerReference(cdc, nodesService, r.scheme); err != nil {
-		return nil, err
-	}
-
 	return nodesService, err
 }
 
 func (r *CassandraDataCenterReconciler) createOrUpdateSeedNodesService(cdc *cassandraoperatorv1alpha1.CassandraDataCenter) (*corev1.Service, error) {
-	seedNodesService := &corev1.Service{ObjectMeta: dataCenterResourceMetadata(cdc, "seeds"),}
+	seedNodesService := &corev1.Service{ObjectMeta: dataCenterResourceMetadata(cdc, "seeds")}
 
 	_, err := controllerutil.CreateOrUpdate(context.TODO(), r.client, seedNodesService, func(_ runtime.Object) error {
 		seedNodesService.Spec = corev1.ServiceSpec{
 			ClusterIP: "None",
 			Ports: []corev1.ServicePort{
```

Comment:
7001 for secure internode

Reply:
Port numbers in services don't really matter when we're using it for Pod discovery -- all Pods can talk to any port. But if you don't specify any ports, the service doesn't work. There was a bug ID, but I've lost it..

```// a port needs to be defined for the service to be resolvable (#there-was-a-bug-ID-and-now-I-cant-find-it)
                        .ports(ImmutableList.of(new V1ServicePort().name("internode").port(7000)))```